kotlinx-coroutines-core/jvm/src/debug/internal/AgentInstallationType.kt

```diff
@@ -3,9 +3,13 @@ package kotlinx.coroutines.debug.internal
 /**
  * Object used to differentiate between agent installed statically or dynamically.
  * This is done in a separate object so [DebugProbesImpl] can check for static installation
- * without having to depend on [kotlinx.coroutines.debug.AgentPremain], which is not compatible with Android.
+ * without having to depend on [AgentPremain], which is not compatible with Android.
  * Otherwise, access to `AgentPremain.isInstalledStatically` triggers the load of its internal `ClassFileTransformer`
  * that is not available on Android.
+ *
+ * Usage Note: Fleet (Reflection): FleetDebugProbes
+ * Usage Note: Android (Hard Coded, ignored for Leak Detection)
+ * Usage Note: IntelliJ (Suppress KotlinInternalInJava): CoroutineDumpState
  */
 internal object AgentInstallationType {
     internal var isInstalledStatically = false
```

Comment:
No sense to mark this as `internal`, as we want this in the BCV dump.

Reply:
`isInstalledStatically` is used from the listed products, rights? Then it, too, should be in the API dump, so not `internal`.

Reply:
```kotlin
@PublishedApi
internal object AgentInstallationType {
    var isInstalledStatically = false
}
```

Results in 
```
public final class kotlinx.coroutines.debug.internal.AgentInstallationType {
  public static final kotlinx.coroutines.debug.internal.AgentInstallationType INSTANCE;
  public final boolean isInstalledStatically();
  public final void setInstalledStatically(boolean);
  static {};
}

```

Reply:
However the original code 
```kotlin
@PublishedApi
internal object AgentInstallationType {
    internal var isInstalledStatically = false
}

```

Results in 
```
public final class kotlinx.coroutines.debug.internal.AgentInstallationType {
  public static final kotlinx.coroutines.debug.internal.AgentInstallationType INSTANCE;
  public final boolean isInstalledStatically$kotlinx_coroutines_core();
  public final void setInstalledStatically$kotlinx_coroutines_core(boolean);
  static {};
}

```

Reply:
Therefore, making it public will break current reflection users
Code example from Fleet
```
  override fun setInstalledStatically(installed: Boolean) {
    c.getDeclaredMethod("setInstalledStatically\$kotlinx_coroutines_core", Boolean::class.java).invoke(instance, installed)
  }
  override fun isInstalledStatically(): Boolean {
    return c.getDeclaredMethod("isInstalledStatically\$kotlinx_coroutines_core").invoke(instance) as Boolean
  }
```